Use string as array

The following program `sample.go` triggers a panic:

```go
package main

const itoa64 = "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"

func main() {
	println(itoa64[2])
}
```

Expected result:
```console
$ go run ./sample.go
48
```

Got:
```console
$ yaegi ./sample.go
scope.go:151: nil type
panic: nil type

goroutine 1 [running]:
log.Panic(0xc0001c12a0, 0x1, 0x1)
	/usr/local/Cellar/go/1.12.5/libexec/src/log/log.go:333 +0xac
github.com/containous/yaegi/interp.(*Scope).add(0xc0000b6480, 0x0, 0x7)
	/Users/marc/go/src/github.com/containous/yaegi/interp/scope.go:151 +0x1c2
github.com/containous/yaegi/interp.(*Interpreter).Cfg.func2(0xc0001eef00)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:580 +0x50d2
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001eef00, 0xc0001c1d50, 0xc0001c1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:113 +0xaf
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001eed00, 0xc0001c1d50, 0xc0001c1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001eec00, 0xc0001c1d50, 0xc0001c1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001eeb00, 0xc0001c1d50, 0xc0001c1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001ee600, 0xc0001c1d50, 0xc0001c1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Node).Walk(0xc0001ee000, 0xc0001c1d50, 0xc0001c1d88)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:110 +0x70
github.com/containous/yaegi/interp.(*Interpreter).Cfg(0xc00017e060, 0xc0001ee000, 0xc0000b40a0, 0x4, 0x0, 0x0, 0x4)
	/Users/marc/go/src/github.com/containous/yaegi/interp/cfg.go:38 +0x1b2
github.com/containous/yaegi/interp.(*Interpreter).Eval(0xc00017e060, 0xc0001ec000, 0x85, 0x285, 0xc0001ec000, 0x85, 0x0, 0xc0001cc1a0)
	/Users/marc/go/src/github.com/containous/yaegi/interp/interp.go:248 +0x135
main.main()
	/Users/marc/go/src/github.com/containous/yaegi/yaegi.go:61 +0x443
```

--- a/_test/str0.go
+++ b/_test/str0.go
@@ -1,0 +1,12 @@
+package main
+
+import "fmt"
+
+const itoa64 = "./0123456789ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz"
+
+func main() {
+	fmt.Printf("%v %T\n", itoa64[2], itoa64[2])
+}
+
+// Output:
+// 48 uint8

--- a/interp/cfg.go
+++ b/interp/cfg.go
@@ -572,9 +572,12 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 		case IndexExpr:
 			wireChild(n)
 			t := n.child[0].typ
-			if t.cat == ValueT {
+			switch t.cat {
+			case ValueT:
 				n.typ = &Type{cat: ValueT, rtype: t.rtype.Elem()}
-			} else {
+			case StringT:
+				n.typ = t
+			default:
 				n.typ = t.val
 			}
 			n.findex = scope.add(n.typ)
@@ -582,10 +585,10 @@ func (interp *Interpreter) Cfg(root *Node) ([]*Node, error) {
 			switch k := t.TypeOf().Kind(); k {
 			case reflect.Map:
 				n.gen = getIndexMap
-			case reflect.Array, reflect.Slice:
+			case reflect.Array, reflect.Slice, reflect.String:
 				n.gen = getIndexArray
 			default:
-				err = n.cfgError("type is not an array, slice or map: %v", t.id())
+				err = n.cfgError("type is not an array, slice, string or map: %v", t.id())
 			}
 
 		case BlockStmt:

